File: springboot/src/main/java/com/edufelip/meer/security/guards/FirebaseAuthGuard.java
package com.edufelip.meer.security.guards;

import com.edufelip.meer.security.SecurityProperties;
import jakarta.servlet.http.HttpServletRequest;

public class FirebaseAuthGuard {
    public static final String AUTH_HEADER = "Authorization";
    private final SecurityProperties props;

    public FirebaseAuthGuard(SecurityProperties props) {
        this.props = props;
    }

    public void validate(HttpServletRequest request) {
        if (props.isDisableAuth()) return;
        String header = request.getHeader(AUTH_HEADER);
        if (header == null || !header.startsWith("Bearer ")) {
            throw new GuardException("Missing or invalid " + AUTH_HEADER + " header");
        }
        // TODO: add Firebase verification
    }
}

File: springboot/src/main/kotlin/com/edufelip/meer/security/guards/FirebaseAuthGuard.kt
package com.edufelip.meer.security.guards

import com.edufelip.meer.security.SecurityProperties
import jakarta.servlet.http.HttpServletRequest

class FirebaseAuthGuard(private val props: SecurityProperties) {
    fun validate(request: HttpServletRequest) {
        if (props.disableAuth) return
        val header = request.getHeader(AUTH_HEADER)
        if (header.isNullOrBlank() || !header.startsWith("Bearer ")) {
            throw GuardException("Missing or invalid $AUTH_HEADER header")
        }
        // NOTE: Stub — plug in Firebase Admin SDK verification when credentials are configured.
    }

    companion object {
        const val AUTH_HEADER = "Authorization"
    }
}
